packages/js/product-editor/src/components/attribute-input-field/attribute-input-field.tsx

```diff
@@ -196,51 +290,28 @@ export const AttributeInputField: React.FC< AttributeInputFieldProps > = ( {
 				getItemProps,
 				getMenuProps,
 				isOpen,
+			}: {
+				items: NarrowedQueryAttribute[];
+				highlightedIndex: number;
+				getItemProps: (
+					options: UseComboboxGetItemPropsOptions< NarrowedQueryAttribute >
+					// eslint-disable-next-line @typescript-eslint/no-explicit-any
+				) => any;
+				getMenuProps: getMenuPropsType;
+				isOpen: boolean;
 			} ) => {
 				return (
 					<Menu getMenuProps={ getMenuProps } isOpen={ isOpen }>
 						{ isLoading ? (
 							<Spinner />
 						) : (
-							renderItems.map( ( item, index: number ) => (
-								<MenuItem
-									key={ item.id }
-									index={ index }
-									isActive={ highlightedIndex === index }
-									item={ item }
-									getItemProps={ ( options ) => ( {
-										...getItemProps( options ),
-										disabled: item.isDisabled || undefined,
-									} ) }
-									tooltipText={
-										item.isDisabled
-											? disabledAttributeMessage
-											: sanitizeSlugName( item.slug )
-									}
-								>
-									{ isNewAttributeListItem( item ) ? (
-										<div className="woocommerce-attribute-input-field__add-new">
-											<Icon
-												icon={ plus }
-												size={ 20 }
-												className="woocommerce-attribute-input-field__add-new-icon"
-											/>
-											<span>
-												{ sprintf(
-													/* translators: The name of the new attribute term to be created */
-													__(
-														'Create "%s"',
-														'woocommerce'
-													),
-													item.name
-												) }
-											</span>
-										</div>
-									) : (
-										item.name
-									) }
-								</MenuItem>
-							) )
+							renderMenuItems(
```

Comment:
Why do you change to function shape instead of a Component one?

I think it should look better if we have something like...

```jsx
<Menu getMenuProps={ getMenuProps } isOpen={ isOpen }>
	<MenuAttributeItems
		renderItems={ renderItems }
		highlightedIndex={ highlightedIndex }
		getItemProps={ getItemProps }
	/>
</Menu>
````

Reply:
I rewrote the function like a component:

```
	const MenuAttributeItems = ( {
		renderItems,
		highlightedIndex,
		getItemProps,
	}: {
		renderItems: NarrowedQueryAttribute[];
		highlightedIndex: number;
		getItemProps: (
			options: UseComboboxGetMenuPropsOptions
		) => getItemPropsType< NarrowedQueryAttribute >;
	} ) => {
		if ( renderItems.length > 0 ) {
			const menuItems = renderItems.map( ( item, index: number ) => (
				<MenuItem
					key={ item.id }
					index={ index }
					isActive={ highlightedIndex === index }
					item={ item }
					getItemProps={ (
						options: UseComboboxGetMenuPropsOptions
					) => ( {
						...getItemProps( options ),
						disabled: item.isDisabled || undefined,
					} ) }
					tooltipText={
						item.isDisabled
							? disabledAttributeMessage
							: sanitizeSlugName( item.slug )
					}
				>
					{ isNewAttributeListItem( item ) ? (
						<div className="woocommerce-attribute-input-field__add-new">
							<Icon
								icon={ plus }
								size={ 20 }
								className="woocommerce-attribute-input-field__add-new-icon"
							/>
							<span>
								{ sprintf(
									/* translators: The name of the new attribute term to be created */
									__( 'Create "%s"', 'woocommerce' ),
									item.name
								) }
							</span>
						</div>
					) : (
						item.name
					) }
				</MenuItem>
			) );

			return <>{ menuItems }</>;
		}

		return (
			<div className="woocommerce-attribute-input-field__no-results">
				{ __( 'Nothing yet. Type to create.', 'woocommerce' ) }
			</div>
		);
	};
```

Take it like just a simple suggestion. Feel free to ignore it.

Reply:
It allows you to inspect the component beyond that I do think the Component notation fits better:

<img width="616" alt="image" src="https://github.com/woocommerce/woocommerce/assets/77539/af942857-6b90-49b1-ab6e-f1d91df889ab">

Reply:
Good idea! I refactored the method `renderMenuItems` in the commits [e8d18e9](https://github.com/woocommerce/woocommerce/pull/41679/commits/e8d18e906bb8608917462434d4e45b7a9028a442) and [55e006a](https://github.com/woocommerce/woocommerce/pull/41679/commits/55e006af3c4f088fdde4dae10ed6290516241381)